slack-notifier/cfn-templates/slack-notifier.json

```diff
@@ -71,49 +73,10 @@
-                "import os \n",
-                "from urllib2 import Request, urlopen, URLError, HTTPError \n",
-
-                "# Setting up logging \n",
-                "logger = logging.getLogger() \n",
-                "logger.setLevel(logging.INFO) \n",
-
-                "# main function \n",
-
-                "def handler(event, context): \n",
-                "  message =  str(event['detail']['eventDescription'][0]['latestDescription'] + \"\\n\\n<https://phd.aws.amazon.com/phd/home?region=us-east-1#/event-log?eventID=\" + event['detail']['eventArn'] + \"|Click here> for details.\") \n",
-                "  json.dumps(message) \n",
-                "  slack_message = { \n",
-                "    'channel': '",
-                     {
-                      "Ref": "SlackChannel"
-                     },
-                "',\n    'text': message \n",
-                "} \n",
-                "  logger.info(str(slack_message)) \n",
-
-                "  req = Request(\"",{"Ref":"HookURL"}, "\", json.dumps(slack_message)) \n",
-                "  try: \n",
-                "   response = urlopen(req) \n",
-                "   response.read() \n",
-                "   logger.info(\"Message posted to %s\", slack_message['channel']) \n",
-                "  except HTTPError as e: \n",
-                "   logger.error(\"Request failed : %d %s\", e.code, e.reason) \n",
-                "  except URLError as e: \n",
-                "   logger.error(\"Server connection failed: %s\", e.reason) \n"
-              ]
-            ]
+            "Fn::Sub": "# Sample Lambda Function to post notifications to a slack channel when an AWS Health event happens\nfrom __future__ import print_function\nimport boto3\nimport json\nimport logging\nimport os\nfrom urllib.request import Request, urlopen, URLError, HTTPError\nfrom urllib.parse import urlencode\n# Setting up logging\nlogger = logging.getLogger()\nlogger.setLevel(logging.INFO)\n# main function\ndef handler(event, context):\n    message =  str(\n        event['detail']['eventDescription'][0]['latestDescription']  +\n        \"\\n\\n<https://phd.aws.amazon.com/phd/home?region=us-east-1#/event-log?eventID=\" +\n        event['detail']['eventArn'] +\n        \"|Click here> for details.\"\n    )\n    json.dumps(message)\n    slack_message = {\n      \"channel\": \"${SlackChannel}\",\n      \"text\": message,\n      \"username\": \"AWS - Personal Health Updates\"\n    }\n    logger.info(str(slack_message))\n    req = Request(\n      \"${HookURL}\",\n      data=json.dumps(slack_message).encode(\"utf-8\"),\n      headers={\"content-type\": \"application/json\"}\n    )\n    try:\n      response = urlopen(req)\n      response.read()\n      logger.info(\"Message posted to: %s\", slack_message['channel'])\n    except HTTPError as e:\n       logger.error(\"Request failed : %d %s\", e.code, e.reason)\n    except URLError as e:\n       logger.error(\"Server connection failed: %s\", e.reason)\n"
```

Comment:
Something went wrong with indentation and formatting here, could you fix before me we merge?

Reply:
I tested and it looks ok, but have a look at the formatting of https://github.com/aws/aws-health-tools/blob/master/slack-notifier/cfn-templates/slack-notifier.yml#L52

Reply:
sure, back to the keyboard in some minutes

Reply:
I'm on it, yaml offers much better multiline string representation , the 1:1 equivalent between yaml and json is what I've suggested here but I will replace "Fn::Sub" for "Fn::Join" for improving readability allowing the previous multiline approach